src/utils/config.ts

```diff
@@ -1,11 +1,33 @@
 import { makeConfig } from 'cleaner-config'
-import { asNumber, asObject, asOptional, asString } from 'cleaners'
+import {
+  asBoolean,
+  asEither,
+  asNumber,
+  asObject,
+  asOptional,
+  asString
+} from 'cleaners'
+
+const asSwapPlugin = asObject({
+  apiKey: asOptional(asString),
+  secret: asOptional(asString),
+  affiliateId: asOptional(asString),
+  affiliateMargin: asOptional(asNumber),
+  partnerContract: asOptional(asString)
+})
+
+const asPlugin = asObject(asEither(asBoolean, asSwapPlugin))
 
 export const asConfig = asObject({
   dbFullpath: asOptional(asString, 'http://username:password@localhost:5984'),
   httpHost: asOptional(asString, '127.0.0.1'),
   httpPort: asOptional(asNumber, 8008),
-  timeBetweenCyclesInMinutes: asOptional(asNumber, 10)
+  timeBetweenCyclesInMinutes: asOptional(asNumber, 10),
```

Comment:
Was this added in anticipation of future needs, or was it added and then never used?

Reply:
Yes, I was planning to use this in the future for the looping the engine.